crates/core/src/blob/repopacker.rs

```diff
@@ -0,0 +1,474 @@
+    repofile::{
+        configfile::ConfigFile,
+        indexfile::{IndexBlob, IndexPack},
+        packfile::PackId,
+    },
+};
+
+use super::{
+    pack_sizer::{DefaultPackSizer, FixedPackSizer, PackSizer},
+    packer::{Packer, PackerStats},
+};
+
+type RawSender = Sender<RusticResult<(Vec<u8>, BlobId, u64, Option<NonZeroU32>)>>;
+
+/// The `Packer` is responsible for packing blobs into pack files.
+///
+/// # Type Parameters
+///
+/// * `BE` - The backend type.
+#[allow(missing_debug_implementations)]
+#[allow(clippy::struct_field_names)]
+#[derive(Clone)]
+pub struct RepositoryPacker {
+    /// The shared indexer containing the backend.
+    indexer: SharedIndexer,
+    /// The sender to send blobs to the packer.
+    sender: Sender<(Bytes, BlobId)>,
+    /// The sender to send raw blobs to the packer.
+    raw_sender: RawSender,
+    /// The receiver to receive the status from the raw packer.
+    finish: Receiver<RusticResult<PackerStats>>,
+}
+
+impl RepositoryPacker {
+    #[allow(clippy::unnecessary_wraps)]
+    pub fn new_with_default_sizer<BE: DecryptWriteBackend>(
+        be: BE,
+        blob_type: BlobType,
+        indexer: SharedIndexer,
+        config: &ConfigFile,
+        total_size: u64,
+    ) -> RusticResult<Self> {
+        let pack_sizer = DefaultPackSizer::from_config(config, BlobType::Data, total_size);
+        Self::new(be, blob_type, indexer, pack_sizer)
+    }
+}
+
+impl RepositoryPacker {
+    /// Creates a new `Packer`.
+    ///
+    /// # Type Parameters
+    ///
+    /// * `BE` - The backend type.
+    ///
+    /// # Arguments
+    ///
+    /// * `be` - The backend to write to.
+    /// * `blob_type` - The blob type.
+    /// * `indexer` - The indexer to write to.
+    /// * `config` - The config file.
+    /// * `total_size` - The total size of the pack file.
+    ///
+    /// # Errors
+    ///
+    /// * If sending the message to the raw packer fails.
+    /// * If converting the data length to u64 fails
+    #[allow(clippy::unnecessary_wraps)]
+    pub fn new<BE: DecryptWriteBackend, S: PackSizer + Send + Sync + 'static>(
+        be: BE,
+        blob_type: BlobType,
+        indexer: SharedIndexer,
+        pack_sizer: S,
+    ) -> RusticResult<Self> {
+        let mut packer = Packer::new(*be.key(), pack_sizer, blob_type);
+
+        let file_writer = Actor::new(
+            FileWriterHandle {
+                be: be.clone(),
+                indexer: indexer.clone(),
+                cacheable: blob_type.is_cacheable(),
+            },
+            1,
+            1,
+        );
+
+        let (tx, rx) = bounded(0);
+        let (tx_raw, rx_raw) = bounded(0);
+        let (finish_tx, finish_rx) = bounded::<RusticResult<PackerStats>>(0);
+        let repository_packer = Self {
+            indexer: indexer.clone(),
+            sender: tx,
+            raw_sender: tx_raw.clone(),
+            finish: finish_rx,
+        };
+
+        let _join_handle = std::thread::spawn(move || {
```

Comment:
Don't we want to use the join handles somewhere and join the threads, e.g. in case we get Ctrl + C or something?